### PR TITLE
chore: adds eslint rule to prefer optional chain

### PR DESCRIPTION
**Pull request by hatpick**
## Description

Adds eslint prefer-optional-chain rule
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-optional-chain.md

## Task Item

#minor

### CODE REVIEW

**Review by beyackle — APPROVED** [review]
Looks good from here. Will VS Code auto-transform these too?

File: Composer/.eslintrc.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ module.exports = {
     '@typescript-eslint/no-empty-function': 'off',
     '@typescript-eslint/no-unnecessary-type-assertion': 'off',
     '@typescript-eslint/no-use-before-define': 'warn',
+    '@typescript-eslint/prefer-optional-chain': 'error',
```
Comment (GeoffCoxMSFT) [inline Composer/.eslintrc.js:35]: Seems odd for a 'prefer' to cause an error.  Should we start with warn?

File: Composer/packages/adaptive-flow/src/adaptive-flow-editor/utils/cursorTracker/locateElement.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export function locateNearestElement(
   filterAttrs?: string[]
 ): SelectorElement {
   // Get elements that meet the filter criteria
-  let elementArr = elements.filter((element) => !filterAttrs || filterAttrs.find((key) => !!element[key]));
+  let elementArr = elements.filter((element) => !filterAttrs || filterAttrs?.find((key) => !!element[key]));
```
Comment (GeoffCoxMSFT) [inline Composer/packages/adaptive-flow/src/adaptive-flow-editor/utils/cursorTracker/locateElement.ts:22]: Given the !filterAttrs || is still there, does this need the .?

File: Composer/packages/adaptive-form/src/components/ErrorInfo.tsx [hunk 3]
```diff
@@ -9,7 +9,7 @@ const ErrorInfo: React.FC<FallbackProps> = ({ componentStack, error }) => (
   <div style={{ marginRight: '20px' }}>
     <MessageBar isMultiline={false} messageBarType={MessageBarType.error}>
       <p style={{ whiteSpace: 'pre-wrap' }}>
-        <strong>{formatMessage({ default: 'An error occured in the form editor!', id: 'ErrorInfo_part1' })}</strong>
+        <strong>{formatMessage({ default: 'An error occurred in the form editor!', id: 'ErrorInfo_part1' })}</strong>
```
Comment (GeoffCoxMSFT) [inline Composer/packages/adaptive-form/src/components/ErrorInfo.tsx:12]: You changed a formatMessage string, so make sure you are checking in an updated en-US.json.

File: Composer/.eslintrc.js (hunk 1 above)
Comment (hatpick) [reply Composer/.eslintrc.js:35]: I "prefer" erro :), save will autofix it so it should be fine.

File: Composer/packages/adaptive-flow/src/adaptive-flow-editor/utils/cursorTracker/locateElement.ts (hunk 2 above)
Comment (hatpick) [reply Composer/packages/adaptive-flow/src/adaptive-flow-editor/utils/cursorTracker/locateElement.ts:22]: this is not my code, it's autofix.

**Comment by hatpick** [comment]
@beyackle yes, it may cause some false positive tho:
_Note: there are a few edge cases where this rule will false positive. Use your best judgement when evaluating reported errors._

**Review by GeoffCoxMSFT — DISMISSED** [review]

**Review by GeoffCoxMSFT — APPROVED** [review]

**Pull request merged**